repair/repair.cc

```diff
@@ -683,6 +684,7 @@ void repair::shard_repair_task_impl::check_failed_ranges() {
 void repair::shard_repair_task_impl::check_in_abort_or_shutdown() {
     try {
         _as.check();
+        service::topology_guard{_topo_guard}.check();
```

Comment:
This check is called on each loop of row level repair. It is wasteful to construct a topology_guard all the time for performance reasons.  How about materializing the topology_guard at the beginning of repair::shard_repair_task_impl::run. 

I think shard_repair_task_impl is a proper place within the task hierarchies to materialize topology_guard.